src/entity.js

```diff
@@ -0,0 +1,73 @@
+function runTransforms(transforms, entry) {
+  const [originalKey, originalValue] = entry
+
+  const transform = transforms.find(({ test }) =>
+    new RegExp(test).test(originalKey)
+  )?.transform
+
+  if (transform == null) return [originalKey, originalValue]
+
+  const result = transform(originalValue, originalKey)
+  if (Array.isArray(result) && result.length <= 2) {
+    const [transfomedValue, transfomedKey] = result
+    return [transfomedKey ?? originalKey, transfomedValue]
+  }
+  return [originalKey, result]
+}
+
+const createProperty = (
+  initialValue,
+  { enumerable = true, configurable = true, writable = true } = {}
+) => {
+  const descriptor = {
+    enumerable,
+    configurable,
+  }
+
+  const value = {
+    initial: initialValue,
+  }
+
+  if (writable) {
```

Comment:
Because I am stupid. 

Also, because I want original values to be immutable. I want to send minimal path to the server when necessary so I want to store modifications in separate place.